### PR TITLE
update ensemblvep and snpeff modules

### PR DESCRIPTION
**Pull request by maxulysse**
## PR checklist

Closes #XXX 

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

**Comment by maxulysse** [comment]
adding the extra_files path as input for vep for people wanting to use let's say CADD plugin for example...

Populate the `vep_extra_files` channel
```
if (params.cadd_wg_snvs && params.cadd_wg_snvs_tbi && params.cadd_indels && params.cadd_indels && params.cadd_indels_tbi) {
    vep_extra_files = Channel.empty().mix(
        Channel.fromPath(params.cadd_wg_snvs),
        Channel.fromPath(params.cadd_wg_snvs_tbi),
        Channel.fromPath(params.cadd_indels),
        Channel.fromPath(params.cadd_indels_tbi)
    ).collect()
} else {
    vep_extra_files = []
}
```
setting up args:
```
        ext.args          = [
            '--everything --filter_common --per_gene --total_length --offline',
            (params.cadd_cache && params.cadd_wg_snvs && params.cadd_indels) ? '--plugin CADD,whole_genome_SNVs.tsv.gz,InDels.tsv.gz' : '',
        ].join(' ').trim()
```

File: tests/modules/ensemblvep/main.nf [hunk 1]
```diff
@@ -10,5 +10,5 @@ workflow test_ensemblvep {
         file(params.test_data['sarscov2']['illumina']['test_vcf'], checkIfExists: true)
     ]
 
-    ENSEMBLVEP ( input, "WBcel235", "caenorhabditis_elegans", "104", [] )
+    ENSEMBLVEP ( input, "WBcel235", "caenorhabditis_elegans", "104", [], [] )
```
Comment (FriederikeHanssen) [inline tests/modules/ensemblvep/main.nf:13]: Are these tests actually passing or throwing errors? Because I can't get them to pass in sarek

**Comment by FriederikeHanssen** [comment]
Does this also work for other plugins or are they too specific?

**Comment by maxulysse** [comment]
Locally:
```
╰─≻TMPDIR=~ PROFILE=docker nextflow run tests/subworkflows/nf-core/annotation/ensemblvep -entry annotation_ensemblvep -c tests/config/nextflow.config
N E X T F L O W  ~  version 22.04.0
Launching `tests/subworkflows/nf-core/annotation/ensemblvep/main.nf` [jovial_lovelace] DSL2 - revision: c37fdf3f9a
executor >  local (2)
[c7/0a5238] process > annotation_ensemblvep:ANNOTATION_ENSEMBLVEP:ENSEMBLVEP (test)       [100%] 1 of 1 ✔
[b0/6287e0] process > annotation_ensemblvep:ANNOTATION_ENSEMBLVEP:TABIX_BGZIPTABIX (test) [100%] 1 of 1 ✔
```
```
╰─≻TMPDIR=~ PROFILE=docker nextflow run tests/subworkflows/nf-core/annotation/snpeff -entry annotation_snpeff -c tests/config/nextflow.config
N E X T F L O W  ~  version 22.04.0
Launching `tests/subworkflows/nf-core/annotation/snpeff/main.nf` [friendly_mercator] DSL2 - revision: 0cae19ccf8
executor >  local (2)
[56/3d3454] process > annotation_snpeff:ANNOTATION_SNPEFF:SNPEFF (test)           [100%] 1 of 1 ✔
[c1/61a3ae] process > annotation_snpeff:ANNOTATION_SNPEFF:TABIX_BGZIPTABIX (test) [100%] 1 of 1 ✔
```

**Comment by maxulysse** [comment]
> Does this also work for other plugins or are they too specific?

trying to add genesplicer as well in sarek, hopefully should work

**Comment by FriederikeHanssen** [comment]
> Locally:
> 
> ```
> ╰─≻TMPDIR=~ PROFILE=docker nextflow run tests/subworkflows/nf-core/annotation/ensemblvep -entry annotation_ensemblvep -c tests/config/nextflow.config
> N E X T F L O W  ~  version 22.04.0
> Launching `tests/subworkflows/nf-core/annotation/ensemblvep/main.nf` [jovial_lovelace] DSL2 - revision: c37fdf3f9a
> executor >  local (2)
> [c7/0a5238] process > annotation_ensemblvep:ANNOTATION_ENSEMBLVEP:ENSEMBLVEP (test)       [100%] 1 of 1 ✔
> [b0/6287e0] process > annotation_ensemblvep:ANNOTATION_ENSEMBLVEP:TABIX_BGZIPTABIX (test) [100%] 1 of 1 ✔
> ```
> 
> ```
> ╰─≻TMPDIR=~ PROFILE=docker nextflow run tests/subworkflows/nf-core/annotation/snpeff -entry annotation_snpeff -c tests/config/nextflow.config
> N E X T F L O W  ~  version 22.04.0
> Launching `tests/subworkflows/nf-core/annotation/snpeff/main.nf` [friendly_mercator] DSL2 - revision: 0cae19ccf8
> executor >  local (2)
> [56/3d3454] process > annotation_snpeff:ANNOTATION_SNPEFF:SNPEFF (test)           [100%] 1 of 1 ✔
> [c1/61a3ae] process > annotation_snpeff:ANNOTATION_SNPEFF:TABIX_BGZIPTABIX (test) [100%] 1 of 1 ✔
> ```

Yes somehow locally (if I attach the MultiQC report) it turns out they actually errored out but don't throw an error to nextflow (if that makes sense). I believe the reason is the mismatch vcf and respective cache organisms. Should be in the .command.err:
```
[cb/39acfc] process > NFCORE_SAREK:SAREK:ANNOTATE:ANNOTATION_SNPEFF:SNPEFF (1234N)                    [100%] 1 of 1, cached: 1 ✔
[2f/53b101] process > NFCORE_SAREK:SAREK:ANNOTATE:ANNOTATION_SNPEFF:ANNOTATION_BGZIPTABIX (1234N)     [100%] 1 of 1, cached: 1 ✔
[45/0efa02] process > NFCORE_SAREK:SAREK:ANNOTATE:ANNOTATION_ENSEMBLVEP:ENSEMBLVEP (1234N)            [100%] 1 of 1, cached: 1 ✔
[85/560f5b] process > NFCORE_SAREK:SAREK:ANNOTATE:ANNOTATION_ENSEMBLVEP:ANNOTATION_BGZIPTABIX (1234N) [100%] 1 of 1, cached: 1 ✔
[b0/297910] process > NFCORE_SAREK:SAREK:CUSTOM_DUMPSOFTWAREVERSIONS (1)                              [100%] 1 of 1 ✔
[cf/87baab] process > NFCORE_SAREK:SAREK:MULTIQC                                                      [100%] 1 of 1, failed: 1 ✔
```

in the `.command.log`:
```
Possible precedence issue with control flow operator at /opt/conda/envs/nf-core-vep-104.3/lib/site_perl/5.26.2/Bio/DB/IndexedBase.pm line 805.
2022-05-05 11:43:47 - INFO: disabling SIFT
2022-05-05 11:43:47 - INFO: disabling PolyPhen
WARNING: Chromosome 1 not found in annotation sources or synonyms on line 1
2022-05-05 11:43:47 - INFO: Disabling --hgvs; using --offline and no FASTA file found
```

and for snpeff in the `.command.err`:
```

ERRORS: Some errors were detected
Error type      Number of errors
ERROR_CHROMOSOME_NOT_FOUND      52

```

File: modules/ensemblvep/meta.yml [hunk 2]
```diff
@@ -47,6 +36,10 @@ input:
       type: file
       description: |
         path to VEP cache (optional)
+  - extra_files:
+      type: tuple
+      description: |
+        path to file(s) needed for plugins  (optional)
```
Comment (jfy133) [inline modules/ensemblvep/meta.yml:42]: single path to a directory containing, or list of paths?

**Review by jfy133 — APPROVED** [review]
Looks OK to me, but never looked at an official nf-core subwokflow

File: modules/ensemblvep/meta.yml (hunk 2 above)
Comment (maxulysse) [reply modules/ensemblvep/meta.yml:42]: either

Comment (maxulysse) [reply modules/ensemblvep/meta.yml:42]: I guess either could work

**Comment by maxulysse** [comment]
> Looks OK to me, but never looked at an official nf-core subwokflow

I think the subworkflows are still a WIP, we need more to actually figure what we want on how we want it.
I think it's a good idea to have small minimal ones like these two as a prototype easy to modify

**Comment by maxulysse** [comment]
> Yes somehow locally (if I attach the MultiQC report) it turns out they actually errored out but don't throw an error to nextflow (if that makes sense). I believe the reason is the mismatch vcf and respective cache organisms. Should be in the .command.err:

I'll look more into that, thanks for explaining the issue

**Comment by FriederikeHanssen** [comment]
> > Yes somehow locally (if I attach the MultiQC report) it turns out they actually errored out but don't throw an error to nextflow (if that makes sense). I believe the reason is the mismatch vcf and respective cache organisms. Should be in the .command.err:
> 
> I'll look more into that, thanks for explaining the issue

arguably very annoying that they don't throw a proper error or nextflow won't pick it up, but the results are not usable

**Pull request merged**